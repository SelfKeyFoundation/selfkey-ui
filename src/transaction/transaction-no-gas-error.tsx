import * as React from 'react';
import injectSheet, { StyleSheet } from 'react-jss';

import { TransactionBox } from './transaction-box';
import { H2 } from '../typography/headings';
import { P } from '../typography/paragraph';
import { A } from '../typography/link';
import { Copy } from '../common/copy';

const styles: StyleSheet = {
	header: {
		textAlign: 'center',
	},

	body: {
		textAlign: 'justify',
	},

	publicKeyWrapper: {
		textAlign: 'center',
	},

	publicKey: {
		fontFamily: 'ProximaNovaRegular',
		color: '#93b0c1',
	},
};

export type TransactionNoGasErrorProps = {
	cryptoCurrency: string;
	publicKey: string;
	closeAction?: ((event: React.MouseEvent<HTMLElement>) => void);
	openLink?: ((link: string) => void);
};

export const TransactionNoGasError = injectSheet(styles)<TransactionNoGasErrorProps>(
	({ classes, children, cryptoCurrency, publicKey, openLink, closeAction }) => {
		const gasExplanationUrl = 'https://help.selfkey.org/article/87-how-does-gas-impact-transaction-speed';
		const handleLinkClick = (event: React.MouseEvent<HTMLElement>) => {
			event.preventDefault();
			if (!openLink) {
				return;
			}
			openLink(gasExplanationUrl);
		};
		return (
			<TransactionBox cryptoCurrency={cryptoCurrency} closeAction={closeAction}>
				<H2 className={classes.header}>Transaction Failed</H2>
				<div className={classes.body}>
					<P>
						You don't have enough Ethereum (ETH) to pay for the network transaction fee. Please transfer
						some ETH to this address and try again. Your ETH address of this wallet is listed below. To
						learn more about transaction fees, click{' '}
						<A href={gasExplanationUrl} onClick={handleLinkClick}>
							here
						</A>
						.
					</P>
					<div className={classes.publicKeyWrapper}>
						<span className={classes.publicKey}>{publicKey}</span>
						<Copy text={publicKey} />
					</div>
				</div>
			</TransactionBox>
		);
	}
);

export default TransactionNoGasError;
