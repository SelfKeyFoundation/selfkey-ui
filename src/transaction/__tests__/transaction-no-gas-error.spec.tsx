import * as React from 'react';
import { create } from 'react-test-renderer';
import { TransactionNoGasError } from '../transaction-no-gas-error';

describe('Transaction no Gas Error', () => {
	const externalUrl = 'https://help.selfkey.org/article/87-how-does-gas-impact-transaction-speed';

	it('shoul open external link', () => {
		const mockFunction = jest.fn();
		const event = {
			preventDefault: jest.fn()
		};
		const component = create(
			<TransactionNoGasError
				openLink={mockFunction}
				publicKey="0x4184288c556524df9cb9e58b73265ee66dca4efe"
			/>
		);

		let tree = component.toJSON();
		const instance = component.root;
		expect(tree).toMatchSnapshot();

		const viewLink = instance.find(el => el.type === 'a' && el.props.onClick);
		viewLink.props.onClick(event);

		expect(mockFunction).toHaveBeenCalledWith(externalUrl);
		expect(event.preventDefault).toHaveBeenCalled();

		tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('shoul not call open external link', () => {
		const mockFunction = jest.fn();
		const event = {
			preventDefault: jest.fn()
		};
		const component = create(
			<TransactionNoGasError publicKey="0x4184288c556524df9cb9e58b73265ee66dca4efe" />
		);

		let tree = component.toJSON();
		const instance = component.root;
		expect(tree).toMatchSnapshot();

		const viewLink = instance.find(el => el.type === 'a' && el.props.onClick);
		viewLink.props.onClick(event);

		expect(mockFunction).not.toHaveBeenCalled();
		expect(event.preventDefault).toHaveBeenCalled();

		tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
