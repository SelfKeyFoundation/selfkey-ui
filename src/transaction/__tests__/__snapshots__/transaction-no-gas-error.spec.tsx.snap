// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction no Gas Error should show Transaction not enough gas error screen 1`] = `
<div
  className="Component-transactionBox-0-1-5"
>
  <div
    className="Component-closeDialogIconWrapper-0-1-8"
    onClick={undefined}
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1 Component-root-0-1-10"
      color={undefined}
      focusable="false"
      viewBox="\\"0 0 38 38"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <circle
          cx="18"
          cy="18"
          fill="#262F39"
          r="18"
          stroke="#303C49"
        />
        <path
          d="M10.815 9.64L9.64 10.814 16.824 18l-7.185 7.185 1.176 1.176L18 19.176l7.185 7.185 1.176-1.176L19.176 18l7.185-7.185-1.176-1.176L18 16.824z"
          fill="#3B4A5A"
          fillRule="nonzero"
        />
      </g>
    </svg>
  </div>
  <div
    className="Component-iconWrapper-0-1-6"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1 Component-root-0-1-11 Component-icon-0-1-7"
      color={undefined}
      focusable="false"
      viewBox="0 0 44 44"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <rect
          fill="#9418DC"
          height="44"
          rx="9"
          width="44"
        />
        <path
          d="M11.37 20.286l10.027-4.1 9.719 4.205L21.382 5.5 11.37 20.286zm.031 4.55l10.012 5.627 10.212-5.627L21.505 38.5 11.401 24.836zm10.043-6.645l10.15 4.385-10.15 5.508L11 22.456l10.444-4.265z"
          fill="#FFF"
          fillRule="nonzero"
        />
      </g>
    </svg>
  </div>
  <div>
    <h1
      className="Component-base-0-1-12 Component-h2-0-1-14 Component-header-0-1-1"
    >
      Transaction Failed
    </h1>
    <div
      className="Component-body-0-1-2"
    >
      <p
        className="undefined Component-p-0-1-15 undefined"
      >
        You don't have enough Ethereum (ETH) to pay for the network transaction fee. Please transfer some ETH to this address and try again. Your ETH address of this wallet is listed below. To learn more about transaction fees, click
         
        <a
          className="undefined Component-a-0-1-16 undefined"
          href="https://help.selfkey.org/article/87-how-does-gas-impact-transaction-speed"
          onClick={[Function]}
        >
          here
        </a>
        .
      </p>
      <div
        className="Component-publicKeyWrapper-0-1-3"
      >
        <span
          className="Component-publicKey-0-1-4"
        >
          0x4184288c556524df9cb9e58b73265ee66dca4efe
        </span>
        <button
          className="CopyComponent-copyButton-0-1-17"
          onClick={[Function]}
        >
          <span
            className="CopyComponent-copyButtonSpan-0-1-18"
          >
            COPY
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
